src/Control/Distributed/Process.hs

```diff
@@ -402,12 +407,23 @@ spawnChannelLocal proc = do
 callLocal ::
      Process a  -- ^ Process to run
   -> Process a  -- ^ Value returned
-callLocal proc = do
-  mv <- liftIO newEmptyMVar
-  self <- getSelfPid
-  _ <- spawnLocal $ do
-     link self
-     try proc >>= liftIO . putMVar mv
-  liftIO $ takeMVar mv >>=
-     either (throwIO :: SomeException -> IO a) return
-
+callLocal proc = do 
```

Comment:
"mask_" the whole thing?

Otherwise, can you link the piece of wisdom that ensures that exceptions cannot happen between the `spawnLocal` and the `takeMVar`? Even if it exists I suspect it is not a feature of the language but something compiler-specific.